pkg/apply/task/apply_task.go

```diff
@@ -128,22 +142,48 @@ func (a *ApplyTask) Start(taskContext *taskrunner.TaskContext) {
 	}()
 }
 
-func (a *ApplyTask) sendTaskResult(taskContext *taskrunner.TaskContext, err error) {
-	taskContext.TaskChannel() <- taskrunner.TaskResult{
-		Err: err,
+func newApplyOptions(eventChannel chan event.Event, strategy common.DryRunStrategy, factory util.Factory) (applyOptions, error) {
+	discovery, err := factory.ToDiscoveryClient()
+	if err != nil {
+		return nil, err
+	}
+	dynamic, err := factory.DynamicClient()
+	if err != nil {
+		return nil, err
 	}
-}
 
-func (a *ApplyTask) setApplyOptionsFields(eventChannel chan event.Event) {
-	if ao, ok := a.ApplyOptions.(*apply.ApplyOptions); ok {
-		ao.DryRun = a.DryRunStrategy.ClientDryRun()
-		ao.ServerDryRun = a.DryRunStrategy.ServerDryRun()
-		adapter := &KubectlPrinterAdapter{
+	emptyString := ""
+	return &apply.ApplyOptions{
+		VisitedNamespaces: sets.NewString(),
+		VisitedUids:       sets.NewString(),
+		Overwrite:         true, // Normally set in apply.NewApplyOptions
+		OpenAPIPatch:      true, // Normally set in apply.NewApplyOptions
+		Recorder:          genericclioptions.NoopRecorder{},
+		IOStreams: genericclioptions.IOStreams{
+			Out:    ioutil.Discard,
+			ErrOut: ioutil.Discard, // TODO: Warning for no lastConfigurationAnnotation
+			// is printed directly to stderr in ApplyOptions. We
+			// should turn that into a warning on the event channel.
+		},
+		DeleteOptions: &delete.DeleteOptions{},
+		PrintFlags: &genericclioptions.PrintFlags{
+			OutputFormat: &emptyString,
+		},
+		ServerSideApply: strategy.ServerDryRun(),
+		FieldManager:    "kubectl", // TODO: Make this configurable
+		DryRun:          strategy.ClientOrServerDryRun(),
+		ServerDryRun:    strategy.ServerDryRun(),
+		ToPrinter: (&KubectlPrinterAdapter{
```

Comment:
Nice.

Reply:
When looking at the `ApplyOptions` in kubectl apply, the `Complete` function has the following code for the `ToPrinter` to append a string for dry-run:
```
  o.ToPrinter = func(operation string) (printers.ResourcePrinter, error) {
        o.PrintFlags.NamePrintFlags.Operation = operation
        if o.DryRun {
            o.PrintFlags.Complete("%s (dry run)")
        }
        if o.ServerDryRun {
            o.PrintFlags.Complete("%s (server dry run)")
        }
        return o.PrintFlags.ToPrinter()
    }
```

Do we need to add this, or is it not used in `kpt live apply` ?

Reply:
We don't use it. We replace it with a different implementation of the `ToPrinter` function.